elephant/functional_connectivity_src/total_spiking_probability_edges.py

```diff
@@ -0,0 +1,266 @@
+    elif isinstance(delay_times, list):
+        pass
+    elif isinstance(delay_times, Iterable):
+        delay_times = list(delay_times)
+
+    NCC_d = np.zeros((len(delay_times), n_neurons, n_neurons))
+
+    for index, delay_time in enumerate(delay_times):
+        # Uses theoretical zero-padding for shifted values,
+        # but since $0 \cdot x = 0$ values can simply be omitted
+        if delay_time == 0:
+            CC = spike_trains_array[:, :] @ spike_trains_array[:, :].transpose()
+
+        elif delay_time > 0:
+            CC = (
+                spike_trains_array[:, delay_time:]
+                @ spike_trains_array[:, :-delay_time].transpose()
+            )
+
+        else:
+            CC = (
+                spike_trains_array[:, :delay_time]
+                @ spike_trains_array[:, -delay_time:].transpose()
+            )
+
+        # Normalize using std deviation
+        NCC = CC / std_factors / n_bins
+
+        # Compute cross correlation at given delay time
+        NCC_d[index, :, :] = NCC
+
+    # Move delay_time axis to back of array
+    # Makes index using neurons more intuitive → (n_neuron, n_neuron, delay_times)
+    NCC_d = np.moveaxis(NCC_d, 0, -1)
+
+    return NCC_d
+
+
+def generate_edge_filter(
+    surrounding_window_size: int,
+    observed_window_size: int,
+    crossover_window_size: int,
+) -> np.ndarray:
+    r"""Generate an edge filter
+
+    The edge filter is generated using following piecewise defined function:
+
+    a = surrounding_window_size
+    b = observed_window_size
+    c = crossover_window_size
+
+    .. math::
+        g_{(i)} = \begin{cases}
+            - \frac{1}{a} & 0 \lt i \leq a \\
+            \frac{2}{b} & a+c \lt i \leq a + b + c \\
+            - \frac{1}{a} & a+b+2c \lt i \leq 2a + b + 2c
+            \end{cases}
+
+    """
+    filter_length = (
+        (2 * surrounding_window_size)
+        + observed_window_size
+        + (2 * crossover_window_size)
+    )
+    i = np.arange(1, filter_length + 1, dtype=np.float64)
+
+    conditions = [
+        (i > 0) & (i <= surrounding_window_size),
+        (i > (surrounding_window_size + crossover_window_size))
+        & (i <= surrounding_window_size + observed_window_size + crossover_window_size),
+        (
+            i
+            > surrounding_window_size
+            + observed_window_size
+            + (2 * crossover_window_size)
+        )
+        & (
+            i
+            <= (2 * surrounding_window_size)
+            + observed_window_size
+            + (2 * crossover_window_size)
+        ),
+    ]
+
+    values = [
+        -(1 / surrounding_window_size),
+        2 / observed_window_size,
+        -(1 / surrounding_window_size),
+        0,
+    ]  # Default Value
+
+    filter = np.piecewise(i, conditions, values)
+
+    return filter
+
+
+def generate_running_total_filter(observed_window_size: int) -> np.ndarray:
+    return np.ones(observed_window_size)
+
+
+class tspe_filter_pair(NamedTuple):
```

Comment:
```suggestion
class TspeFilterPair(NamedTuple):
```
Naming convention for classes is CamelCase